### DISCUSSION TITLE
Reorder ClusterTree

### DISCUSSION THREAD

**Discussion by annaliem** [satijalab/seurat#5231] [Q&A]
Is there a way to reorder the cluster tree in ```PlotClusterTree``` to the original cluster order?
```
BuildClusterTree(SO, slot = "scale.data")
PlotClusterTree(SO, direction = "rightwards")
```
![image](https://user-images.githubusercontent.com/58446857/138785543-9249aba3-93cf-49b7-92e3-197f5afca01c.png)

Looking to have the tree ordered as OT1, OT2, OT3, OT4, etc. so I can stick it next to a dot plot. Using Seurat 4.0.3.

**Comment by AmelZulji** [comment]
Hi @annaliem, 

One way to have "synchronized" dotplot and clustertree is to reorder identities based on the clustertree:
`SO <– BuildClusterTree(SO, slot = "scale.data", reorder = T)`
Then order of identities in you object will match order of clustertree plot and you can merge both plots. 

hope it helps,
Amel

**Reply by annaliem** [reply]
Hi Amel,

Thank you for your response! Unfortunately I have tried that and I only see reordering of one or two clusters. I apologize - I should have included that in my initial post.

```
BuildClusterTree(SO, slot = "scale.data", assay = "SCT", reorder = T)
tree <- Tool(object = SO, slot = "BuildClusterTree")
ape::plot.phylo(x = tree, direction = "rightwards")
```
![image](https://user-images.githubusercontent.com/58446857/139309218-cdc37b7f-1dfb-4f53-a00b-c174b8dd5e2b.png)

Note: I used ```ape::plot.phylo``` to plot without nodes, but ordering between that and ```PlotClusterTree``` is identical.

**Reply by AmelZulji** [reply]
Hi @annaliem, 
One possible reason that you are seeing unexpected results is because you are not assigning the built tree to the object.  Instead of this:
`BuildClusterTree(SO, slot = "scale.data", assay = "SCT", reorder = T)`

you should be doing this:
`SO <- BuildClusterTree(SO, slot = "scale.data", assay = "SCT", reorder = T)`

Using `pbmc_small` object provided within Seurat package, this is how it should look like: 

```
library(Seurat)

# Rename clusters randomly
pbmc_small <- RenameIdents(pbmc_small, 
                           "2" = "cluster1",
                           "0" = "cluster2",
                           "1" = "cluster3")

# Build cluster tree without reordering
pbmc_small <- BuildClusterTree(pbmc_small)
tree <- Tool(object = pbmc_small, slot = "BuildClusterTree")
ape::plot.phylo(x = tree, direction = "rightwards")
```
![git1](https://user-images.githubusercontent.com/49162517/140582456-75f703ae-490a-411c-b0a4-0feb1281cb0c.png)

If you plot the dotplot, the order of clusters in dotplot does not match the order of clusters in the tree


```
# Plot some random genes
DotPlot(pbmc_small, features = c("MS4A1", "CCL4"))
```
![git2](https://user-images.githubusercontent.com/49162517/140582754-91c06e07-f6cb-4318-ad80-155bdd445580.png)

However, if you build the tree with reordering then the orders of cluster in dotplot will match the order from the tree: 
```
pbmc_small <- BuildClusterTree(pbmc_small, reorder = T)
DotPlot(pbmc_small, features = c("MS4A1", "CCL4"))
```
![git3](https://user-images.githubusercontent.com/49162517/140583384-1592a255-7e74-4292-94e7-732226e98cb1.png)

Then you can you can plot those 2 plots side by side and the order will match. 

Hope it helps.

BEst,
Amel
